### ISSUE TITLE
Bug: `indent` with character encoded as two code units

### ISSUE THREAD

**Issue by regseb** [eslint/eslint#16127]
### Environment

Node version: v18.4.0
npm version: 8.13.2
Local ESLint version: 8.19.0
Global ESLint version: none
Operating System: Ubuntu 22.04 LTS

### What parser are you using?

Default (Espree)

### What did you do?

[Demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IGluZGVudDogW1wiZXJyb3JcIiwgMiwge1wiRnVuY3Rpb25EZWNsYXJhdGlvblwiOiB7XCJwYXJhbWV0ZXJzXCI6IFwiZmlyc3RcIn19XSAqL1xuXG5mdW5jdGlvbiBhKGZvbyxcbiAgICAgICAgICAgYmFyKSB7XG4gIHJldHVybiBcImJhelwiO1xufVxuXG5mdW5jdGlvbiDwkIyYKGZvbyxcbiAgICAgICAgICAgYmFyKSB7XG4gIHJldHVybiBcImJhelwiO1xufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=) on ESLint Playground.

<details>
<summary>Configuration</summary>

```JSON
{
    "parserOptions": {
        "ecmaVersion": "latest",
        "sourceType": "module"
    },
    "rules": {
        "indent": [2,  2, {"FunctionDeclaration": {"parameters": "first"}}]
    }
}
```
</details>

```js
function 𐌘(foo,
           bar) {
  return "baz";
}
```

### What did you expect to happen?

No error reported.

### What actually happened?

> Expected indentation of 12 spaces but found 11.  ([indent](https://eslint.org/docs/rules/indent))

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

The problem is similar to the issue **#15914**. JavaScript uses [UTF-16](https://en.wikipedia.org/wiki/UTF-16#Description) for `String`:

- A character before _U+FFFF_ is encoded as **one** code unit.
- A character after _U+FFFF_ is encoded as **two** code units.

[`String.length`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/length#unicode) counts code units instead of characters.

**Comment by nzakas** [comment]
Thanks. Related:
https://github.com/acornjs/acorn/issues/342

**Comment by nzakas** [comment]
Hi folks, I just wanted to let everyone know that the ESLint team has decided to deprecate all formatting rules, including indent, and recommend people use source code formatters like [Prettier](https://prettier.io) or [dprint](https://dprint.dev) instead. As a result, we are closing all outstanding issues related to these rules. For more information, please see https://github.com/eslint/eslint/issues/17522. Thanks.
